classes/battle/BattleManager.php

```diff
@@ -736,6 +752,25 @@ protected function setupFighterAttack(Fighter $fighter, LegacyFighterAction $act
             $attack->raw_damage = 0;
         }
 
+        // Setup clash effects
+        foreach ($attack->jutsu->effects as $effect) {
+            switch ($effect->effect) {
+                case 'piercing':
+                    $fighter->piercing_percent += $effect->effect_amount / 100;
+                    break;
+                case 'substitution':
+                    $fighter->substitution_percent += $effect->effect_amount / 100;
+                    break;
+                case 'counter':
+                    $fighter->counter_percent += $effect->effect_amount / 100;
+                    break;
+                case 'immolate':
+                    $fighter->immolate_percent += $effect->effect_amount / 100;
```

Comment:
Hmm... I'm inclined to keep this in the jutsu object itself and apply the effects in collision function rather than setting as a global on the fighter, if they are not multi-turn status effects